packages/relay-compiler/transforms/ValidateRequiredArgumentsTransform.js

```diff
@@ -110,7 +110,11 @@ function validateRequiredArgumentsOnNode(
 ): void {
   const nodeArgsSet = new Set(node.args.map(arg => arg.name));
   for (const arg of definitionArgs) {
-    if (schema.isNonNull(arg.type) && !nodeArgsSet.has(arg.name)) {
+    if (
+      arg.defaultValue === undefined &&
```

Comment:
nit:  `arg.defaultValue == null` here

Reply:
Thanks, @josephsavona - updated and rebased. Please have a look again.